### ISSUE TITLE
Show() works on desktop but not on mobile

### ISSUE THREAD

**Issue by tmlmt** [atomiks/tippyjs#922]
## Bug description

`<popup_div_dom_element>._tippy.show()` doesn't work on mobile device

### My setup

I use tippy.js together with an OpenLayers map, as part of an application bundled with webpack. 
I have checked this on my iPhone (iOS 14), but also by using the Web Developer tool of Firefox 86 (Desktop) 

### Why this is weird

- The same code works perfectly fine on desktop
- I have no other error nor warning in the console and npm/webpack builds the code without error
- When emulating a smartphone, and when I manually type in the Web Console ```popup = document.getElementbyId('popup')
popup._tippy.show()``` then the tooltip shows up!
- When outputting the _tippy object before the `show()` command, the object exists, and has registered the new content changed in the previous line with `setContent()`

## Reproduction

I tried to reproduce the bug, but didn't manage. Had to simplify a lot but I'm essentially doing what's in this pen, and ... which works as intended ...
https://codepen.io/tmlmt/pen/eYBPjBg
(Open with a mobile, or with a Web Developer tool emulating a mobile device, for the events to be triggered)

## EDIT 

- I actually discovered issue #133 and @zanzlanz's async trick worked ! So I am able to move on, but this is still a bug IMO. 
- This triggered me to search a bit about this setTimeout function, and it turns out there might be an explanation [here](https://stackoverflow.com/questions/779379/why-is-settimeoutfn-0-sometimes-useful). Could be that the DOM doesn't have time to update after the `setContent()` command and showing immediately after, with a mobile browser engine?

**Comment by atomiks** [comment]
I think you want to use `touch: "hold"` instead. There might be something conflicting in the default mobile behavior which causes `hide()` to be called immediately after it shows.

```js
tippy('#target', {
  content: 'tooltip',
  touch: 'hold',
  triggerTarget: document.querySelector('#trigger')
});
```

Possibly setting `hideOnClick: false` will work if you're adding the listeners manually, although, there may be other issues in that case.

**Comment by tmlmt** [comment]
Thanks for looking into this @atomiks.`touch: 'hold'` didn't work, but `hideOnClick: false` worked, in the sense that the tooltip shows on mobile without `setTimeout`. The thing is that now ... the tooltips doesn't hide on click on desktop. 

 I think we're closing in. So indeed, it seems that `hide()` is called immediately after it shows? It maybe due to the way [mobile browsers simulate tap events](https://patrickhlauke.github.io/touch/tests/results/). Changing my eventlistener to listen to `click` (last one in the list of simulated events) instead of `pointerup` as I was doing before doesn't solve things. But I know that my code reaches `show()`. Would it be that my event listener _and then_ tippy's listener cancel themselves?

One thing I can add it that the tooltip is supposed to appear so that the clicked / touched pixel ends up _on_ the tooltip itself.

**Comment by atomiks** [comment]
> touch: 'hold' didn't work

So you want to tap the screen instead of pressing+holding it? 

In the case of touchstart, click gets fired afterward so that's why it's immediately closing I believe, unless you hold your finger down instead of letting go immediately like a tap/click. I'm just wondering why you're using `touchstart` instead of the default listeners in this case?

**Comment by tmlmt** [comment]
Oh sorry! I confused you with my initial CodePen (which failed to reproduce the case that I encounter in my actual setup with OpenLayers, etc.). I have updated it to now reflect a bit closer what it's looking like in reality, but I still can't reproduce the bug ...
I am not using `pointerstart` in my code: I've been using `pointerup`, but as I said in my previous message, I've just tried with `click` in my app and it equally failed to show the tooltip.
